lib/capistrano/itamae/dsl.rb

```diff
@@ -49,6 +49,7 @@ def ssh_options(server)
         options << "--user #{ssh_options[:user]}" if ssh_options[:user]
         options << "--port #{ssh_options[:port]}" if ssh_options[:port]
         options << "--key #{ssh_options[:key]}" if ssh_options[:key]
+        options << "--dry-run" if dry_run?
```

Comment:
`dry_run?` is available since `capistrano` v3.5.0+
https://github.com/capistrano/capistrano/blob/master/CHANGELOG.md#350

Please add `>= 3.5.0` to `add_dependency "capistrano"`